code/game/objects/items.dm

```diff
@@ -244,8 +244,9 @@
 
 /obj/item/throw_impact(atom/hit_atom)
 	..()
-	if(drop_sound)
-		playsound(src, drop_sound, 50, 0)
+	for(var/mob/M in player_list)
+		if(drop_sound && (M.client.prefs.asfx_togs & ASFX_DROP))
+			playsound(src, drop_sound, 50, 0)

```

Comment:
Doesn't this play the sound once for every player?

Reply:
That is a good question. I _believe_ the answer is yes, but only if in range to hear it, as the function should be. But I have no idea of the reality of it.

Reply:
Well, the point I was moreso making is that. `playsound()` plays a sound to _all_ mobs within range. So basically, by doing this, you play a sound to all mobs in range _for_ every mob that has the preference toggled on. So like, a power of two thing. This means that the check should be handled within `playsound`.

Now, if we go over to `playsound`, we discover that it already does something similar to this with footsteps and ambiance. So a quick fix would be to add a, `is_drop_sound` argument to the proc list and place the check in there. _However_, this is also a short sighted fix and will result in the pooling of arguments for that proc ad infinitum.

A more elegant fix would be to modify the signature of the `playsound` proc to look as follows:
```dm
/proc/playsound(atom/source, soundin, vol as num, vary, extrarange as num, falloff, is_global, usepressure = 1, environment = -1, required_preferences = 0, required_asfx_toggles = 0)
```

For footsteps, we'd invoke:
```dm
playsound(src, T.footstep_sound, 70, 1, required_asfx_toggles = ASFX_FOOTSTEPS)
```
For ambience:
```dm
playsound(src, T.footstep_sound, 70, 1, required_preferences = SOUND_AMBIENCE)
```
For drop sounds:
```dm
playsound(src, T.footstep_sound, 70, 1, required_asfx_toggles = ASFX_DROPSOUNDS)
```

To implement this, within the proc it would have to look something like this:
```dm
/proc/playsound(atom/source, soundin, vol as num, vary, extrarange as num, falloff, is_global, usepressure = 1, environment = -1, required_preferences = 0, required_asfx_toggles = 0)
	// ...
	for (var/P in player_list)
		var/mob/M = P
		if(!M || !M.client)
			continue

		var/distance = get_dist(M, turf_source)
		if(distance <= (world.view + extrarange) * 3)
			var/turf/T = get_turf(M)

			// These checks are split into multiple ifs for readability reasons.

			if (!T || T.z != turf_source.z)
				continue

			if (required_preferences && (M.client.prefs.toggles & required_preferences) != required_preferences)
				continue

			if (required_asfx_toggles && (M.client.prefs.asfx_togs & required_asfx_toggles) != required_asfx_toggles)
				continue
			// ...
```

This also removes my earlier requirement for a proc to fetch the preferences.